graphs/fleurys_algorithm.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+
+
+class Graph:
+    def __init__(self, vertices) -> None:
+        self.v = vertices
+        self.graph = defaultdict(list)
+
+    def add_edge(self, u, v) -> None:
+        """
+        Function to add edges to the graph
+        """
+        self.graph[u].append(v)
+        self.graph[v].append(u)
+
+    def rmv_edge(self, u, v) -> None:
+        """
+        Remove u-v edge from the graph
+        """
+        for index, key in enumerate(self.graph[u]):
+            if key == v:
+                self.graph[u].pop(index)
+        for index, key in enumerate(self.graph[v]):
+            if key == u:
+                self.graph[v].pop(index)
+
+    def dfs_count(self, v, visited) -> int:
+        """
+        A DFS based function to count reachable vertices from v
+        """
+        count = 1
+        visited[v] = True
+        for i in self.graph[v]:
+            if not visited[i]:
+                count = count + self.dfs_count(i, visited)
+        return count
+
+    def is_valid_next_edge(self, u, v) -> bool:
+        """
+        The function to check if edge u-v can be considered as next edge in
+         Euler Tour
+        """
+
+        if len(self.graph[u]) == 1:
+            return True
+        else:
+            visited = [False] * self.v
+            count1 = self.dfs_count(u, visited)
+
+            self.rmv_edge(u, v)
+            visited = [False] * self.v
+            count2 = self.dfs_count(u, visited)
+
+            self.add_edge(u, v)
+
+            return count1 <= count2
+
+    def print_euler_util(self, u) -> None:
+        """
+        To print the elular tour
+        """
+        for v in self.graph[u]:
+            if self.is_valid_next_edge(u, v):
+                print("%d-%d " % (u, v)),
```

Comment:
```suggestion
                print(f"{u}-{v}"),
```
f-strings as discussed in CONTRIBUTING.md.